cooperator/__init__.py

```diff
@@ -1,3 +1,13 @@
 from . import models
 from . import report
 from . import wizard
+
+from odoo import api, SUPERUSER_ID
+
+
+def _assign_default_mail_template_ids(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    mail_template_fields = env["res.company"]._get_cooperator_mail_template_fields()
+    for field, xmlid in mail_template_fields.items():
+        company_ids = env["res.company"].search([(field, "=", False)])
+        company_ids._assign_default_cooperator_mail_template(field, xmlid)
```

Comment:
there is a duplication of logic between here and `res.company.create()`. it would be better to create a method on `res.company` that sets all this data (if it is not set yet) and that would be called here and from `res.company.create()`. moreover, it is a little strange here to loop over the fields and then search for companies for each of them inside the loop. i something like this would feel less strange:
```python
env.companies._setup_default_mail_templates()
```

Reply:
I did consider this, but I was on the fence about making the changes to facilitate that. The chief difference between what happens in `res.company` and what happens in the hook is the filtering of companies depending on whether the field is truthy or not. i.e. don't overwrite existing fields

I could move that check to `_assign_default_cooperator_mail_template`, and your proposal would work. That method currently _would_ override existing fields. I wasn't sure whether to change that.

As a compromise, I'll add an `override` boolean kwarg, and then re-jig this a little bit.

Reply:
done